Задания спринта №8 реализованы.

Maksim-Isaev opened the pull request:
В ходе работы выполнены ДЗ №1, №2, №3. 

Функционал перехода из main_activity  в другие activity  реализован согласно ДЗ№2, т.е
1.  Переход в  Search_activity через анонимный класс
2.  Переход в Media_activity через лямбда лямбда-выражение
3.  Переход в Settings_activity через анонимный класс.

Реализован функционал тулбара в каждом activity (Search, Media, Settings)

Maksim-Isaev committed:
Sprint 8. Homework 1. Added new branch - DEV. Added Main and Settings layout

Maksim-Isaev committed:
Sprint 8. Homework 1. Added new branch - DEV. Added Main and Settings layout

Maksim-Isaev committed:
Sprint 8. Homework 1. Optimize Main layout

Maksim-Isaev committed:
Sprint 8. Homework 2. Clicking on the buttons on the main screen is implemented using various methods. The first button uses an anonymous class to implement its functionality, the second uses a lambda expression, and the third also uses an anonymous class.

Maksim-Isaev committed:
Работа над дебагом

Maksim-Isaev committed:
Sprint 8. HomeWork 3. Fixed bugs. Added functional a transition between the Main, Search, Media, and Settings screens.

faritowich commented on app/src/main/java/com/practicum/playlistmaker/MainActivity.kt:
```diff
@@ -1,20 +1,42 @@
+
+        val button1 = findViewById<Button>(R.id.button1)
+        val button2 = findViewById<Button>(R.id.button2)
+        val button3 = findViewById<Button>(R.id.button3)
```
Не очень хороший нейминг для переменных - он должен отражать суть содержимого, а button3, button2 итд её не отражают. Лучше что-то типа buttonSearch, buttonSettings

faritowich commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,19 +1,77 @@
+
+
+    <Button
+        android:id="@+id/button1"
```
аналогично другому комменту про нейминг

faritowich commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,19 +1,77 @@
+        android:text="@string/search"
+        android:textAllCaps="false"
+        android:textColor="@color/black"
+        android:textSize="@dimen/text_normal"
```
Все размеры текстов задаются при помощи sp. dp - для остальных размеров. В теории про это написано, стоит повторить

faritowich commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,19 +1,77 @@
+        app:cornerRadius="16dp"
+        app:icon="@drawable/ic_search"
+        app:iconGravity="textStart"
+        app:iconPadding="8dp"
```
Много повторяющихся ресурсов в проекте не вынесено

faritowich commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,108 @@
+            android:layout_height="wrap_content"
+            android:layout_gravity="end"
+            android:src="@drawable/ic_galochka"
+            android:textSize="20sp" />
```
Для чего у ImageView атрибут textSize? Надо убрать везде

faritowich commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,108 @@
+            android:layout_gravity="start"
+            android:text="@string/Agreement"
+            android:textColor="@color/black_setting_menu"
+            android:textSize="@dimen/text_16" />
```
Текст в dp - надо поменять на sp

Maksim-Isaev committed:
Sprint 8. HomeWork 3.1.
Corrections have been made according to the comments:
The resources contain the dimensions that are used in the layout several times;
The name of the buttons has been renamed to MainActivity;
The text size values have been adjusted to "SP";
and other minor corrections.

Maksim-Isaev commented:
Исправления согласно замечаниям внесены в свежем коммите.
